app/controllers/users_controller.rb

```diff
@@ -71,6 +71,18 @@ def due_for_judgement
     @predictions = @predictions.not_private unless current_user == @user
     @predictions = @predictions.select { |x| x.due_for_judgement? }
   end
+  
+  def generate_api_token
+    @user = User.find_by_login(params[:login])
+    
+    if @user && @user.update_attributes(api_token: User.generate_api_token)
+      flash[:notice] = "Generated a new API token!"
+    else
+      flash[:error]  = "We couldn't generate a new API token, sorry."
```

Comment:
When providing an error message like this it's good to keep in mind what it would be like as the user to receive the message. In this case it does not give them any idea what to do to remedy the situation (I'm, aware that the other messages in this controller aren't great but may as well not make it worse).

First of all I would suggest adding `generate_api_token` to the [list of actions that require login](https://github.com/jaysonvirissimo/predictionbook/blob/cleaner-api/app/controllers/users_controller.rb#L3). You can then assume that `current_user` will be set with the logged in user and use it instead of `@user`. The implementation as it stands would allow anyone to generate a new API token for any user they know the username of.

With that change made you know that it was the `update_attributes` call that failed, which should provide errors. So the message could become something along these lines:

```ruby
"Unable to generate new API token due to these errors: #{current_user.errors.full_messages.to_sentence}. Please ensure your user profile is complete"
```